src/llama_recipes/finetuning.py

```diff
@@ -228,12 +238,29 @@ def main(**kwargs):
             use_kahan_summation=False,
             weight_decay=train_config.weight_decay,
         )
+    elif optimizer_in_backward_available:
+        print(f"setting up optimizer overlap")
+        optim_kwargs = {"lr": train_config.lr}
+        _apply_optimizer_in_backward(
+            optimizer_class=optim.AdamW,
```

Comment:
If the optimizer_in_backward_available flag is set and user has selected AnyPrecisionAdamW, will be good to add for that case as well, unless there is a restriction on which optimizers support this feature.

Reply:
Added for Anyprecision as well, as per my test it does not break.